extmod/cyw43_config_common.h

```diff
@@ -0,0 +1,104 @@
+
+// The board-level config will be included here, so it can set some CYW43 values.
+#include <stdio.h>
+#include "py/mpconfig.h"
+#include "py/mperrno.h"
+#include "py/mphal.h"
+#include "py/runtime.h"
+#include "extmod/modnetwork.h"
+#include "lwip/apps/mdns.h"
+#include "pendsv.h"
+
+// This file is included at the top of port-specific cyw43_configport.h files,
+// and holds the MicroPython-specific but non-port-specific parts.
+//
+// It's included into both some MicroPython sources and the lib/cyw43-driver sources.
+
+#define CYW43_IOCTL_TIMEOUT_US          (1000000)
+#define CYW43_NETUTILS                  (1)
+#define CYW43_PRINTF(...)               mp_printf(MP_PYTHON_PRINTER, __VA_ARGS__)
+
+#define CYW43_EPERM                     MP_EPERM // Operation not permitted
+#define CYW43_EIO                       MP_EIO // I/O error
+#define CYW43_EINVAL                    MP_EINVAL // Invalid argument
+#define CYW43_ETIMEDOUT                 MP_ETIMEDOUT // Connection timed out
+
+#define CYW43_THREAD_ENTER              MICROPY_PY_LWIP_ENTER
+#define CYW43_THREAD_EXIT               MICROPY_PY_LWIP_EXIT
+#define CYW43_THREAD_LOCK_CHECK
+
+#define CYW43_HOST_NAME                 mod_network_hostname_data
+
+#define CYW43_ARRAY_SIZE(a)             MP_ARRAY_SIZE(a)
+
+#define CYW43_HAL_PIN_MODE_INPUT        MP_HAL_PIN_MODE_INPUT
+#define CYW43_HAL_PIN_MODE_OUTPUT       MP_HAL_PIN_MODE_OUTPUT
+#define CYW43_HAL_PIN_PULL_NONE         MP_HAL_PIN_PULL_NONE
+#define CYW43_HAL_PIN_PULL_UP           MP_HAL_PIN_PULL_UP
+#define CYW43_HAL_PIN_PULL_DOWN         MP_HAL_PIN_PULL_DOWN
+
+#define CYW43_HAL_MAC_WLAN0             MP_HAL_MAC_WLAN0
+
+#define cyw43_hal_ticks_us              mp_hal_ticks_us
+#define cyw43_hal_ticks_ms              mp_hal_ticks_ms
+
+#define cyw43_hal_pin_obj_t             mp_hal_pin_obj_t
+#define cyw43_hal_pin_config            mp_hal_pin_config
+#define cyw43_hal_pin_read              mp_hal_pin_read
+#define cyw43_hal_pin_low               mp_hal_pin_low
+#define cyw43_hal_pin_high              mp_hal_pin_high
+
+#define cyw43_hal_get_mac               mp_hal_get_mac
+#define cyw43_hal_get_mac_ascii         mp_hal_get_mac_ascii
+#define cyw43_hal_generate_laa_mac      mp_hal_generate_laa_mac
+
+#define cyw43_schedule_internal_poll_dispatch(func) pendsv_schedule_dispatch(PENDSV_DISPATCH_CYW43, func)
+
+// Note: this function is only called if CYW43_POST_POLL_HOOK is defined in cyw43_configport.h
+void cyw43_post_poll_hook(void);
+
+static inline void cyw43_delay_us(uint32_t us) {
+    uint32_t start = mp_hal_ticks_us();
+    while (mp_hal_ticks_us() - start < us) {
+    }
+}
+
+static inline void cyw43_delay_ms(uint32_t ms) {
+    // PendSV may be disabled via CYW43_THREAD_ENTER, so this delay is a busy loop.
+    uint32_t us = ms * 1000;
+    uint32_t start = mp_hal_ticks_us();
+    while (mp_hal_ticks_us() - start < us) {
+        mp_event_handle_nowait();
+    }
+}
+
+#define CYW43_EVENT_POLL_HOOK mp_event_handle_nowait()
```

Comment:
For `stm32` and `mimxrt` this was `MICROPY_EVENT_POLL_HOOK` which does something different, especially in the `stm32` port (see the old one in stm32 port). The same for the delay functions which called this macro. I know that eventually we'll use the new functions, but I'd suggest you leave this macro and the delay functions in the port config files for now, to avoid changing the behavior, unless it's time to remove this macro.

Reply:
Oh good catch, I mis-remembered the macro compatibility as being different! Will fix.

Reply:
I've swapped this around so stm32 should be using the same hook as before.

mimxrt should be using mostly the same hooks, although it previously had `cyw43_delay_us` and `cyw43_delay_ms` aliased directly to `mp_hal_delay_us` and `mp_hal_delay_ms` and now it uses the inline function definitions instead. This seems like either a no-op or a small improvement, but I'm not able to test it to verify.